libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -362,10 +362,20 @@ public Bundle createNewAccount(String accountName, String username, String refre
         }
         Account acc = new Account(accountName, getAccountType());
         accountManager.addAccountExplicitly(acc, SalesforceSDKManager.encrypt(refreshToken), new Bundle());
-
-        // Caching auth token otherwise the first call to accountManager.getAuthToken will go to the AuthenticatorService which will do a refresh
-        // That is problematic when the refresh token is set to expire immediately
-        accountManager.setAuthToken(acc, AccountManager.KEY_AUTHTOKEN, SalesforceSDKManager.encrypt(authToken));
+        final Account[] accounts = getAccounts();
+        int numAuthenticatedUsers = accounts == null ? 0 : accounts.length;
+        boolean isFirstUserOrNotIDPFlow = !SalesforceSDKManager.getInstance().isIDPAppLoginFlowActive()
+                || (numAuthenticatedUsers <= 1);
+
+        /*
+         * Sets auth token only if this user is the first user being logged in or NOT an IDP login.
+         * Caching auth token otherwise the first call to 'accountManager.getAuthToken()' will go
+         * to the AuthenticatorService which will do a refresh. That is problematic when the
+         * refresh token is set to expire immediately.
+         */
+        if (isFirstUserOrNotIDPFlow) {
+            accountManager.setAuthToken(acc, AccountManager.KEY_AUTHTOKEN, SalesforceSDKManager.encrypt(authToken));
```

Comment:
Prevents user switching when adding user from IDP context, unless it's the first user being logged in to the app.